lib/internal/test_runner/runner.js

```diff
@@ -98,7 +98,8 @@ function createTestFileList(patterns) {
     cwd,
     exclude: (name) => name === 'node_modules',
   });
-  const results = glob.globSync();
+  // We need to resolve paths as this function's cwd may differ from the process cwd
+  const results = ArrayPrototypeMap(glob.globSync(), (result) => relative(process.cwd(), resolve(cwd, result)));
```

Comment:
I thought we were trying to avoid using `process.cwd()` here?

Reply:
I tried implementing with absolute paths, but it causes many tests to fail as it changes the output of the cli.  
By using a relative path from the current `process.cwd` to the absolute paths resolved from glob patterns, with the provided `cwd` as the root path, we can maintain the same output while still supporting the resolution of test files in a different working directory provided via parameter.

What do you think about this?

I didn't see this problem in the previous implementation because absolute resolution occurred only when the parameter was provided.  
However, I don't like the idea of having two different behaviors or outputs depending on whether the `cwd` parameter was provided

Reply:
I think the issue might be that we are spawning child processes without the `cwd` option. We have also landed support for running processes in the current process, so we would need to make sure that case works as well.